### PR TITLE
Minor fixes

### PR DESCRIPTION
**Pull request by melver**
Not quite sure what the best strategy for handling the sl*b_def.h mess is. We might be better off not touching those files. WDYT?

### CODE REVIEW

File: mm/kfence/report.c [hunk 1]
```diff
@@ -131,7 +131,8 @@ void kfence_report_error(unsigned long address, const struct kfence_metadata *me
 	int num_stack_entries = stack_trace_save(stack_entries, KFENCE_STACK_DEPTH, 1);
 	int skipnr = get_stack_skipnr(stack_entries, num_stack_entries, type);
 
-	lockdep_assert_held(&meta->lock);
+	if (meta)
```
Comment (ramosian-glider) [inline mm/kfence/report.c:135]: This isn't the only place dereferencing `meta` in this function, shall we do anything about the rest?

File: lib/Kconfig.kfence [hunk 2]
```diff
@@ -5,6 +5,7 @@ menuconfig KFENCE
 	bool "KFENCE: low-overhead use-after-free and buffer-overflow detector"
 	depends on HAVE_ARCH_KFENCE && (!KASAN || EXPERT) && (SLAB || SLUB)
 	depends on JUMP_LABEL # To ensure performance, require jump labels
+	select STACKTRACE
```
Comment (ramosian-glider) [inline lib/Kconfig.kfence:8]: Didn't know one could unselect it.

Comment (melver) [reply lib/Kconfig.kfence:8]: Yeah, make tinyconfig did not select it. (Some eventual random config would have hit this and we'd have gotten test-robot complaints.)

File: mm/kfence/report.c (hunk 1 above)
Comment (melver) [reply mm/kfence/report.c:135]: Only for OOB it's non-optional. I added a precondition check.

**Pull request merged**